mitiq/pec/tests/test_channels.py

```diff
@@ -0,0 +1,202 @@
+            conj_sum = sum([k.conj().T @ k for k in kraus_ops])
+            assert np.allclose(conj_sum, np.eye(2 ** num_qubits))
+
+
+def test_matrix_to_vector():
+    for d in [1, 2, 3, 4]:
+        mat = np.random.rand(d, d)
+        assert matrix_to_vector(mat).shape == (d ** 2,)
+        assert (vector_to_matrix(matrix_to_vector(mat)) == mat).all
+
+
+def test_vector_to_matrix():
+    for d in [1, 2, 3, 4]:
+        vec = np.random.rand(d ** 2)
+        assert vector_to_matrix(vec).shape == (d, d)
+        assert (matrix_to_vector(vector_to_matrix(vec)) == vec).all
+
+
+def test_non_squared_dimension():
+    with raises(ValueError, match="must be a square number"):
+        vector_to_matrix(np.random.rand(7))
+    with raises(ValueError, match="must be a square number"):
+        choi_to_super(np.random.rand(7))
+
+
+def test_kraus_to_super():
+    """Tests the function on random channels acting on random states.
+    Channels and states are non-physical, but this is irrelevant for the test.
+    """
+    for num_qubits in (1, 2, 3, 4, 5):
+        d = 2 ** num_qubits
+        fake_kraus_ops = [
+            np.random.rand(d, d) + 1.0j * np.random.rand(d, d)
+            for _ in range(7)
+        ]
+        super_op = kraus_to_super(fake_kraus_ops)
+        fake_state = np.random.rand(d, d) + 1.0j * np.random.rand(d, d)
+        result_with_kraus = sum(
+            [k @ fake_state @ k.conj().T for k in fake_kraus_ops]
+        )
+        result_with_super = vector_to_matrix(
+            super_op @ matrix_to_vector(fake_state)
+        )
+        assert np.allclose(result_with_kraus, result_with_super)
+
+
+def test_super_to_choi():
+    for noise_level in [0, 0.3, 1]:
+        super_damping = kraus_to_super(amplitude_damping_kraus(noise_level, 1))
+        # Apply Pauli Y to get some complex numbers
+        super_op = np.kron(channel(Y)[0], channel(Y)[0].conj()) @ super_damping
+        choi_state = super_to_choi(super_op)
+        # expected result
+        q = LineQubit(0)
+        choi_expected = _operation_to_choi(
+            [AmplitudeDampingChannel(noise_level)(q), Y(q)]
+        )
+        assert np.allclose(choi_state, choi_expected)
+
+
+def test_choi_to_super():
+    # Note: up to normalization, choi_to_super is equal to super_to_choi
+    # and therefore we just run the following consistency test.
+    for dim in (2, 4, 8, 16):
+        rand_mat = np.random.rand(dim ** 2, dim ** 2)
+        assert np.allclose(super_to_choi(choi_to_super(rand_mat)), rand_mat)
+        assert np.allclose(choi_to_super(super_to_choi(rand_mat)), rand_mat)
+
+
+def test_kraus_to_choi():
+    for dim in (2, 4, 8, 16):
+        rand_kraus_ops = [np.random.rand(dim, dim) for _ in range(7)]
+        super_op = kraus_to_super(rand_kraus_ops)
+        expected_choi = super_to_choi(super_op)
+        assert np.allclose(kraus_to_choi(rand_kraus_ops), expected_choi)
+
+
+def test_tensor_product():
```

Comment:
Would be good to have tests for 1 and 0 elements. 1 element just returns the element, and 0 elements throws an IndexError, which I think is appropriate behavior.

Reply:
I added a TypeError if no arguments are given. This is the type of error that e.g. `np.kron()` or `sum()` would give.